### ISSUE TITLE
Inconsistent simplifying of boolean expressions

### ISSUE THREAD

**Issue by piotr-napadlek** [josdejong/mathjs#3172]
Hi, I'm trying to work with evaluating and simplifying boolean and arithmetic expressions. Sometimes my expressions contain additional parameters which values are not known at the time of evaluation, therefore if they are detected I want to store my expression as a simplified string that will be further evaluated once the parameters are known. I tried to achieve this by comparing the results of `evaluate` and `simplify` results whereby evaluate receives all the possible parameters with default values in the scope argument. If the result is equal (after coercing results) I know that the expression does not have any parameters.

This is working fine for working with arithmetic expressions, ie. `1 + 1` will give me `2` with evaluate and `"2"` with simplify. However when working with boolean expressions things get tricky as simplify yields 1s and 0s in results. Have a look at the following examples:

**Describe the bug**
```
evaluate("1 > 0 or 2 < 3").toString(); // returns "true"
simplify("1 > 0 or 2 < 3").toString(); // returns "1"

evaluate("1 > 0 and 2 < 3").toString(); // returns "true"
simplify("1 > 0 and 2 < 3").toString(); // returns "1 and 1" (??)
simplify("1 > 0 and not (2 < 3)").toString(); // returns "0"
```

**To Reproduce**
Run the code above.

Is this expected behaviour? Is there any way to configure the way boolean expressions are resolved by the simplify function? And do I understand correct that there is a bug with simplifying `and` operator sometimes resulting in an expression not fully simplified?

**Comment by josdejong** [comment]
Thanks for reporting.

You see you want to detect additional parameters. Doing so implicitly by comparing `evaluate` and `simplify` sounds tricky to me. You can also detect unknown variables by first parsing an expression with `math.parse`, then use the method `traverse` to find all symbol nodes, and then check if the symbols are known symbols (like `pi`, `sqrt`, `meter`) or unknown symbols.

Docs: https://mathjs.org/docs/expressions/expression_trees.html -> search for method "traverse"
Check whether a variable is known or not: https://github.com/josdejong/mathjs/discussions/2865


You have good points about the need to improve `simplify`. I see two action points:

1. We'll need to look into why `simplify` returns a `"1"` as a result of a logical operator like `or` and relational operators like `>` and `<`, I would expect it to return `true` or not evaluate the result. 
2. It looks like we need to extend `simplify` to evaluate `and` too, like `or`.

Help debugging this would be welcome.

**Comment by dvd101x** [comment]
Hi, this is not much of a debugging but something that has not been stated is that currently simplifying "true" returns "1" and simplifying "false" returns "0".

```js
math.simplify("true").toString()       // returns "1"
math.simplify("false").toString()      // returns "0"
```
So this not only affects logical operators like `or`, `and`, `>`, `<`, etc.

Tested on V9, V10, V11 and V12

**Comment by josdejong** [comment]
Good point David

**Comment by piotr-napadlek** [comment]
Hi, thanks @josdejong and @dvd101x for clarification, yes with this explanation it makes sense and if that is expected then the only thing left unclear is why
```
simplify("true and true").toString(); // returns "1 and 1"
```
I can work with the sugestion from @josdejong for my problem but out of curiosity I will try to understand how is `and` behaving different in simplifying than `or`.

**Comment by josdejong** [comment]
See the simplification logic was missing a case. Fixed now via a41def836077f09b4ba0ccf953e600527ef4720f

**Comment by josdejong** [comment]
Fixed now in `12.4.1`
